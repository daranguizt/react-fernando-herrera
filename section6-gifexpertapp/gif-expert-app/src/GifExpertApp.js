import React, { Fragment, useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifExpertApp = () => {
  //const categories = ["One Punch Man", "Samurai X", "Dragon Ball"];

  const [categories, setCategories] = useState(["One Punch Man"]);

  //   const handleAdd = () => {
  //     setCategories([...categories, "Hunter x Hunter"]);
  //     //setCategories(cat => [...cat, 'Hunter x Hunter']); donde cat es el estado actual
  //   };

  return (
    <Fragment>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      {/* <button onClick={handleAdd}>Agregar</button>*/}

      <div className="allResults">
        {categories.map((item) => (
          <GifGrid key={item} category={item} />
        ))}
      </div>
    </Fragment>
  );
};

export default GifExpertApp;
