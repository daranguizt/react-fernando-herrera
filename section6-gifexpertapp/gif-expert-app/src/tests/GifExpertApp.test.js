import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('tests on GifExpertApp', () => {
    test('should render GifExpertApp and match snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show a list of categories', () => {
        const categories = ['mr popo', 'shaquille'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});