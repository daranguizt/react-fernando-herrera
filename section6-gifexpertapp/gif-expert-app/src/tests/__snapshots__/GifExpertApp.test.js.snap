// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on GifExpertApp should render GifExpertApp and match snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div
    className="allResults"
  >
    <GifGrid
      category="One Punch Man"
      key="One Punch Man"
    />
  </div>
</Fragment>
`;

exports[`tests on GifExpertApp should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div
    className="allResults"
  >
    <GifGrid
      category="mr popo"
      key="mr popo"
    />
    <GifGrid
      category="shaquille"
      key="shaquille"
    />
  </div>
</Fragment>
`;
